tests/legacy-cli/e2e/tests/misc/safari-15-class-properties.ts

```diff
@@ -0,0 +1,43 @@
+import { expectFileToExist, readFile, writeFile } from '../../utils/fs';
+import { ng } from '../../utils/process';
+import { updateJsonFile } from '../../utils/project';
+
+export default async function () {
+  await updateJsonFile('angular.json', (workspace) => {
+    const build = workspace.projects['test-project'].architect.build;
+    build.defaultConfiguration = undefined;
+    build.options = {
+      ...build.options,
+      optimization: false,
+      outputHashing: 'none',
+    };
+  });
+
+  // Matches two types of static fields that indicate that the Safari bug
+  // may still occur. With the workaround this should not appear in bundles.
+  //   - static { this.ecmp = bla }
+  //   - static #_ = this.ecmp = bla
+  const staticIndicatorRegex = /static\s+(\{|#[_\d]+\s+=)/;
```

Comment:
This is a sanity test. Obviously it's not validating using AST, nor validating Safari. Should be helpful regardless though.